ui/controlgroup.js

```diff
@@ -0,0 +1,182 @@
+			"checkboxradio": "input[type='checkbox'], input[type='radio']",
+			"selectmenu": "select"
+		},
+		direction: "horizontal",
+		excludeInvisible: true,
+		classes: {}
+	},
+
+	_create: function() {
+		this._enhance();
+	},
+
+	// The support the enhanced option in jQuery Mobile, we isolate DOM manipulation
+	_enhance: function() {
+		this.element.attr( "role", "toolbar" );
+		this.refresh();
+	},
+
+	_destroy: function() {
+		var that = this;
+		$.each( this.options.items, function( widget, selector ) {
+			that.element.children( selector ).map( function() {
+				return $( this )[ widget ]( "widget" ).removeData( "ui-controlgroup-data" )[ 0 ];
+			} ).removeData( "ui-controlgroup-data" );
+		} );
+		this._callChildMethod( "destroy" );
+		this.element.removeAttr( "role" );
+	},
+
+	_callChildMethod: function( method ) {
+		var that = this;
+
+		this.buttons = $();
+		$.each( this.options.items, function( widget, selector ) {
+			var options = {};
+			if ( that[ "_" + widget + "_options" ] ) {
+				options = that[ "_" + widget + "_options" ]( "middle" );
+			}
+			if ( $.fn[ widget ] && selector ) {
+				that.element
+					.find( selector )[ widget ]( method ? method : options )
+						.each( function() {
+							if ( method !== "destroy" ) {
+								var button = $( this )[ widget ]( "widget" ).data( "ui-controlgroup-data", {
+									"widgetType": widget,
+									"element": $( this )
+								} );
+								that.buttons = that.buttons.add( button );
+							}
+						} );
+			}
+		} );
+	},
+
+	_button_options: function( position, direction ) {
+		return {
+			classes: {
+				"ui-button": {
+					"middle": null,
+					"first": "ui-corner-" + ( direction ? "top" : "left" ),
+					"last": "ui-corner-" + ( direction ? "bottom" : "right" )
+				}[ position ]
+			}
+		};
+	},
+
+	_checkboxradio_options: function( position, direction ) {
+		return {
+			classes: {
+				"ui-checkboxradio-label": {
+					"middle": null,
+					"first": "ui-corner-" + ( direction ? "top" : "left" ),
+					"last": "ui-corner-" + ( direction ? "bottom" : "right" )
+				}[ position ]
+			}
+		};
+	},
+
+	_selectmenu_options: function( position, direction ) {
+		return {
+			width: "auto",
+			classes: {
+				middle: {
+					"ui-selectmenu-button-open": null,
+					"ui-selectmenu-button-closed": null
+				},
+				first: {
+					"ui-selectmenu-button-open":
+					"ui-corner-" + ( direction ? "top" : "tl" ),
+					"ui-selectmenu-button-closed":
+					"ui-corner-" + ( direction ? "top" : "left" )
+				},
+				last: {
+					"ui-selectmenu-button-open":
+					direction ? null : "ui-corner-tr",
+					"ui-selectmenu-button-closed":
+					"ui-corner-" + ( direction ? "bottom" : "right" )
+				}
+
+			}[ position ]
+		};
+	},
+
+	_setOption: function( key, value ) {
+		var original = this.options[ key ];
+
+		this._super( key, value );
+		if ( key === "direction" ) {
+			this._removeClass( "ui-controlgroup-" + original );
```

Comment:
Reminder to self should this `return` here also don't see a reason to run refresh if this is all thats changing?